### PR TITLE
Feature/add google tagmanager connector

### LINKED ISSUE #4254: Add Google Tag Manager Connector
As is available for Google Analytics, DNN should have a standard option to connect a website to Google Tag Manager too. Preferably working identical to the Google Analytics connector, so it will be as easy as possible for site administrators.

### PR DESCRIPTION
**Pull request by skamphuis**
Closes #4254

## Summary
I created this Google Tag Manager connector by simply taking a copy of the Google Analytics Connector project and changing it to fit the needs of Google Tag Manager.
I had to add the Controller class and the Engine class to the project. These are located in the DNN core for Google Analytics. (But I didn't want to change the core)

SiteAnalytics.config will need to be edited too, with the part from the .config file in this PR.

### CODE REVIEW

File: DNN_Platform.build [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
-    <WebsitePath>$(RootDirectory)\Website</WebsitePath>
+    <WebsitePath>D:\Websites\dev\dev-972\files</WebsitePath>
```
Comment (valadas) [inline DNN_Platform.build:3]: Please don't hardcode this path.

File: DNN_Platform.sln [hunk 2]
```diff
@@ -536,6 +540,8 @@ Global
 		{3CD5F6B8-8360-4862-80B6-F402892DB7DD}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{3CD5F6B8-8360-4862-80B6-F402892DB7DD}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{3CD5F6B8-8360-4862-80B6-F402892DB7DD}.Debug|x86.ActiveCfg = Debug|Any CPU
+		{3CD5F6B8-8360-4862-80B6-F402892DB7DD}.Debug-GTM|Any CPU.ActiveCfg = Debug|Any CPU
+		{3CD5F6B8-8360-4862-80B6-F402892DB7DD}.Debug-GTM|x86.ActiveCfg = Debug|Any CPU
```
Comment (valadas) [inline DNN_Platform.sln:544]: Can the multiple Debug-GTM be removed from the solution

File: DNN Platform/Connectors/GoogleTagManager/Dnn.GoogleTagManagerConnector.sln [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+MinimumVisualStudioVersion = 10.0.40219.1
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Dnn.GoogleTagManagerConnector", "Dnn.GoogleTagManagerConnector.csproj", "{ADB57FFD-677C-4C30-9723-6EE0A475315B}"
+EndProject
+Global
+	GlobalSection(SolutionConfigurationPlatforms) = preSolution
+		Cloud_Debug|Any CPU = Cloud_Debug|Any CPU
+		Cloud_Release|Any CPU = Cloud_Release|Any CPU
+		Debug|Any CPU = Debug|Any CPU
+		Release|Any CPU = Release|Any CPU
+	EndGlobalSection
```
Comment (valadas) [inline DNN Platform/Connectors/GoogleTagManager/Dnn.GoogleTagManagerConnector.sln:14]: The module itself should only have a project file and be included in the main solution file, can this solution file be removed.

**Review by valadas — CHANGES_REQUESTED** [review]
Nice job, thanks for contributing this, From a quick look it looks about right to me except the 3 notes I left and then I would like to test that build

File: DNN_Platform.sln (hunk 2 above)
Comment (skamphuis) [reply DNN_Platform.sln:544]: Yes, that was me trying to need a full platform build.

File: DNN_Platform.build (hunk 1 above)
Comment (skamphuis) [reply DNN_Platform.build:3]: Sorry about that. I should be able to use the facepalm emoticon for this.

File: DNN Platform/Connectors/GoogleTagManager/Dnn.GoogleTagManagerConnector.sln (hunk 3 above)
Comment (skamphuis) [reply DNN Platform/Connectors/GoogleTagManager/Dnn.GoogleTagManagerConnector.sln:14]: Sure.

**Comment by skamphuis** [comment]
> 
> 
> Nice job, thanks for contributing this, From a quick look it looks about right to me except the 3 notes I left and then I would like to test that build

Thanks. I corrected my mistakes as you suggested.

**Comment by david-poindexter** [comment]
This is also related to #3155 and if this PR is accepted, we should close that issue as well.  Awesome @skamphuis - I will review the PR as soon as possible.

File: DNN Platform/Connectors/GoogleTagManager/GoogleTagManager.config [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+			</ScriptTemplate>
+		</AnalyticsEngine>
+		<AnalyticsEngine>
+			<EngineType>DotNetNuke.Services.Analytics.GoogleTagManagerEngine, DNN.Connectors.GoogleTagManager</EngineType>
+			<ElementId>Body</ElementId>
+			<InjectTop>True</InjectTop>
+			<ScriptTemplate>
+				<![CDATA[     
+					<!-- Google Tag Manager (noscript) --> 
+					<noscript><iframe src="https://www.googletagmanager.com/ns.html?id=GTM-5SXBKKT"
```
Comment (david-poindexter) [inline DNN Platform/Connectors/GoogleTagManager/GoogleTagManager.config:27]:
```suggestion
					<noscript><iframe src="https://www.googletagmanager.com/ns.html?id=[GTM_ID]"
```

**Review by david-poindexter — CHANGES_REQUESTED** [review]
This looks really good - thank you @skamphuis 🎉

File: DNN_Platform.build (hunk 1 above)
Comment (david-poindexter) [reply DNN_Platform.build]: 🤦

**Review by valadas — APPROVED** [review]
Looks good to me once David suggestion is commited.

I looked at the CI build and the package is there properly.

**Review by david-poindexter — APPROVED** [review]
Awesomeness - I am excited about this one - thanks @skamphuis 🎉

**Pull request merged**